### PR TITLE
remove ReadFileCache and add comonad instance

### PR DESCRIPTION
**Pull request by safareli**

### CODE REVIEW

File: src/Quasar/Advanced/QuasarAF.purs [hunk 1]
```diff
@@ -134,15 +134,15 @@ readFile
   → Maybe Pagination
   → QuasarAFCE JArray
 readFile mode path pagination =
-  map (_.content <<< unwrap) <$> readFileDetail mode path pagination
+  map extract <$> readFileDetail mode path pagination
```
Comment (Cmdv) [inline src/Quasar/Advanced/QuasarAF.purs]: didn't even know that existed, (looks on pursuit for `extract`) 🏃

File: src/Quasar/QuasarF.purs [hunk 2]
```diff
@@ -50,24 +52,29 @@ derive instance functorExpiredContent ∷ Functor ExpiredContent
...
 instance traversableExpiredContent ∷ Traversable ExpiredContent where
   traverse f (ExpiredContent {content, expired}) =
     map (ExpiredContent <<< {expired, content: _}) (f content)
   sequence ec = sequenceDefault ec
 
+instance extendExpiredContent :: Extend ExpiredContent where
+  extend f t@(ExpiredContent { expired, content }) = ExpiredContent { expired, content: f t }
+
+instance comonadExpiredContent :: Comonad ExpiredContent where
+  extract (ExpiredContent { content }) = content
```
Comment (Cmdv) [inline src/Quasar/QuasarF.purs]: would of never got these, not even sure how comonads work yet!

**Comment by safareli** [comment]
Here are couple articles about comonads I have reed before, might be useful:
http://joneshf.github.io/programming/2015/12/31/Comonads-Monoids-and-Trees.html
https://yow.eventer.com/yow-lambda-jam-2015-1305/cofun-with-cofree-comonads-by-david-laing-1891
https://www.youtube.com/watch?v=mqCsfYERzzE
https://bartoszmilewski.com/2017/01/02/comonads/
http://dlaing.org/cofun/posts/free_and_cofree.html

**Comment by Cmdv** [comment]
I've been reading Haskell from first principles book again, currently on Applicative. So will definitely take a look at those articles see if I can get my head around comonads 😄
Cheers again 👍

**Pull request merged**